pyomo/contrib/parmest/parmest.py

```diff
@@ -581,9 +585,9 @@ def _Q_opt(self, ThetaVals=None, solver="ef_ipopt",
 
             model = stsolver.make_ef()
             stream_solver = True
-            ipopt = SolverFactory('ipopt')
-            sipopt = SolverFactory('ipopt_sens')
-            kaug = SolverFactory('k_aug')
+            ipopt = SolverFactory('ipopt', executable=self.solver_path)
+            sipopt = SolverFactory('ipopt_sens', executable=self.solver_path)
+            kaug = SolverFactory('k_aug', executable=self.solver_path)
```

Comment:
I am not sure that this is valid, as the executable specifies the absolute path to the executable (file).  Wouldn't we need to be able to specify a separate executable flag for ipopt, sipopt and kaug?

Would a better / easier approach be to pass an actual solver object into the method (and leave it up to the user to create it)?  If the solver was `None`, you could still create one using default options.

Reply:
I agree with John. It would be better to ask the user to pass solver objects.

Reply:
Or maybe sound the retreat. We are adding a lot complication for something that can be accomplished by changing the PATH.